src/Serilog.AspNetCore/AspNetCore/RequestLoggingMiddleware.cs

```diff
@@ -57,29 +59,31 @@ public async Task Invoke(HttpContext httpContext)
                 await _next(httpContext);
                 var elapsedMs = GetElapsedMilliseconds(start, Stopwatch.GetTimestamp());
                 var statusCode = httpContext.Response.StatusCode;
-                LogCompletion(httpContext, collector, statusCode, elapsedMs, null);
+                await LogCompletion(httpContext, collector, statusCode, elapsedMs, null);
             }
             catch (Exception ex)
-                // Never caught, because `LogCompletion()` returns false. This ensures e.g. the developer exception page is still
-                // shown, although it does also mean we see a duplicate "unhandled exception" event from ASP.NET Core.
-                when (LogCompletion(httpContext, collector, 500, GetElapsedMilliseconds(start, Stopwatch.GetTimestamp()), ex))
             {
+                await LogCompletion(httpContext, collector, 500, GetElapsedMilliseconds(start, Stopwatch.GetTimestamp()), ex);
```

Comment:
This is a regression - the `when` clause executes before unwinding, which we use in order to capture `LogContext` state before it's popped off as `using` blocks are exited.